test/integration_test/storkctl_migrationschedule_test.go

```diff
@@ -0,0 +1,323 @@
+	migrationScheduleName := "automation-test-migration-schedule"
+	var testResult = testResultFail
+	runID := testrailSetupForTest(testrailID, &testResult)
+	defer updateTestRail(&testResult, testrailID, runID)
+	defer migrationScheduleCleanup(t, migrationScheduleName, migrationScheduleNs)
+
+	factory := storkctl.NewFactory()
+	var outputBuffer bytes.Buffer
+	cmd := storkctl.NewCommand(factory, os.Stdin, &outputBuffer, os.Stderr)
+	cmdArgs := []string{"create", "migrationschedule", migrationScheduleName}
+	// add the custom args to the command
+	for key, value := range args {
+		cmdArgs = append(cmdArgs, "--"+key)
+		if value != "" {
+			cmdArgs = append(cmdArgs, value)
+		}
+	}
+	cmd.SetArgs(cmdArgs)
+	// execute the command
+	logrus.Infof("The storkctl command being executed is %v", cmdArgs)
+	if err := cmd.Execute(); err != nil {
+		logrus.Errorf("Storkctl execution failed: %v", err)
+		return
+	}
+	// Get the captured output as a string
+	actualOutput := outputBuffer.String()
+	logrus.Infof("Actual output is: %s", actualOutput)
+	expectedOutput := fmt.Sprintf("MigrationSchedule %v created successfully\n", migrationScheduleName)
+	require.Equal(t, expectedOutput, actualOutput)
+
+	// Validate the created resource
+	specFile := "specs/storkctl-specs/migrationschedule/" + specFileName
+	err := ValidateMigrationScheduleFromFile(t, specFile, migrationScheduleName, migrationScheduleNs)
+	require.NoError(t, err, "Error validating the created resource")
+
+	// If we are here then the test has passed
+	testResult = testResultPass
+	logrus.Infof("Test status at end of %s test: %s", t.Name(), testResult)
+}
+
+func ValidateMigrationScheduleFromFile(t *testing.T, specFilePath string,
+	migrationScheduleName string, migrationScheduleNs string) error {
+	data, err := getByteDataFromFile(specFilePath)
+	if err != nil {
+		return err
+	}
+	migrationSchedule := &storkv1.MigrationSchedule{}
+	dec := yaml.NewYAMLOrJSONDecoder(bytes.NewReader(data), len(data))
+	if err := dec.Decode(&migrationSchedule); err != nil {
+		return err
+	}
+	if err := ValidateMigrationSchedule(t, migrationSchedule, migrationScheduleName, migrationScheduleNs); err != nil {
+		return err
+	}
+	return nil
+}
+
+func ValidateMigrationSchedule(t *testing.T, migrationSchedule *storkv1.MigrationSchedule,
+	migrationScheduleName string, migrationScheduleNs string) error {
+	// We want to validate if the created schedule policy resource matches our expectations
+	actualMigrationSchedule, err := storkops.Instance().GetMigrationSchedule(migrationScheduleName, migrationScheduleNs)
+	if err != nil {
+		logrus.Errorf("Unable to get the created migration schedule")
+		return err
+	}
+	logrus.Info("Trying to validate the migration schedule")
+	if migrationSchedule.Annotations != nil || actualMigrationSchedule.Annotations != nil {
+		require.Equal(t, migrationSchedule.Annotations, actualMigrationSchedule.Annotations, "MigrationSchedule Annotations mismatch")
```

Comment:
nit...This can panic...Can change it to something like this?
```
if migrationSchedule.Annotations != nil {
require.NotNil(t, actualMigrationSchedule.Annotations, "Expected actualMigrationSchedule not to be nil")
require.Equal(t, migrationSchedule.Annotations, actualMigrationSchedule.Annotations, "MigrationSchedule Annotations mismatch")
} else{
require.Nil(t, actualMigrationSchedule.Annotations, ""Expected actualMigrationSchedule not to be nil"")
}
```

Reply:
Thanks for the input. Will address